### DISCUSSION TITLE
How to run own CT data in the demo 01_intro_walnut_conebeam.ipynb?

### DISCUSSION THREAD

**Discussion by TianSong1991** [TomographicImaging/CIL-Demos#135] [Q&A] [answered]
The Cil module is really good to use. Now I want to reconstruct my own data. I used 01_intro_walnut_conebeam.ipynb in the demo to reconstruct it. It has been successfully reconstructed. Now I want to reconstruct my own data.  I copied my own data to the imported walnut data `data.array`, and then modified the pixel_num_v and pixel_num_h attributes of data.geometry. My projection number is only 360, and the angle is from 0 to 359 degrees. I assigned it to `data.geometry.angles` I am not allowed to modify,  and I am not allowed to modify `data.geometry.num_projections`. How can I solve it, or how can I create my own geometry, thank you!

**Comment by TianSong1991** [comment]
![image](https://github.com/TomographicImaging/CIL-Demos/assets/33627940/421bffce-8da8-4462-bccd-3216069add54)
Can you give me a detailed explanation about Angles 0-20, but I don’t quite understand what the meaning of the 20 radians.

**Reply by lauramurgatroyd** [reply]
This is just printing out the first 20 angles (indexes 0 to 19 in the angles list), and shows them in radians rather than degrees

**Reply by TianSong1991** [reply]
Ok ,thank you very much! I see.

**Comment by MargaretDuff** [comment, marked as answer]
Hi @TianSong1991  if you are interested in creating your own geometry, I would recommend having  a look through the notebook: TomographicImaging/CIL-Demos/demos/1_Introduction/exercises/03_where_is_my_reader.ipynb. This takes you through setting up your own geometries.  For more information, see the documentation e.g. https://tomographicimaging.github.io/CIL/nightly/framework.html#configure-the-geometry. 

I would recommend that you join the CIL discord channel https://discord.gg/xXUsCS59. On Friday between 10 and 11 we will be answering user questions and are available for a video call on discord

**Reply by lauramurgatroyd** [reply]
[TomographicImaging/CIL-Demos/demos/1_Introduction/00_CIL_geometry.ipynb](https://github.com/TomographicImaging/CIL-Demos/blob/main/demos/1_Introduction/00_CIL_geometry.ipynb) also provides an introduction to CIL Geometries

**Comment by TianSong1991** [comment]
@MargaretDuff @lauramurgatroyd Thank you for your detailed reply, I will take a look at the introduction and use it in detail, this is a very useful CT reconstruction module.

**Comment by TianSong1991** [comment]
I reconstruct my own data successfully. Thanks for your guiding.

**Reply by lauramurgatroyd** [reply]
You're welcome, glad it worked!
